### ISSUE TITLE
function_score query doesn't have filter property

### ISSUE THREAD

**Issue by GlenRSmith** [elastic/elasticsearch-net#1418]
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-function-score-query.html

The "filter" property isn't available.

![nofilter](https://cloud.githubusercontent.com/assets/1327920/7774145/2639db70-0077-11e5-969d-237a0d36d1d9.gif)

**Comment by GlenRSmith** [comment]
Sample query:

```
curl "http://localhost:9200/test_score/test_type/_search?pretty" -d '{
    "query": {
        "function_score": {
            "filter": {
                "query": {
                    "multi_match": {
                        "query": "John Wayne",
                        "fields": ["title", "credits"]
                    }
                }
            },
            "functions": [
                {
                    "filter": {
                        "query": {
                            "match": {"title": "John Wayne"}
                        }
                    },
                    "weight": 1
                },
                {
                    "filter": {
                        "query": {
                            "match": {"credits": "John Wayne"}
                        }
                    },
                    "weight": 1
                }
            ],
            "score_mode": "sum"
        }
    }
}'
```

**Comment by gmoskovicz** [comment]
@gmarz / @Mpdreamz Maybe we do have a way to do this? Thanks :)

**Comment by gmarz** [comment]
@GlenRSmith thanks for reporting this.

@gmoskovicz Looks like it's simply missing.  It's just a matter of adding `IFilterContainer` to `IFunctionScoreQuery` and all the supporting code that goes with it.  Just like [IQueryContainer](https://github.com/elastic/elasticsearch-net/blob/develop/src/Nest/DSL/Query/FunctionScoreQueryDescriptor.cs#L20).

**Comment by gmarz** [comment]
Closed via #1419

**Comment by GlenRSmith** [comment]
@gmoskovicz Thanks for the rapid response!
